src/base/bittorrent/customstorage.cpp

```diff
@@ -35,6 +35,190 @@
 #include "base/utils/fs.h"
 #include "common.h"
 
+#if (LIBTORRENT_VERSION_NUM >= 20000)
+#include <libtorrent/session.hpp>
+
+std::unique_ptr<lt::disk_interface> customDiskIOConstructor(
+        lt::io_context &ioContext, const lt::settings_interface &settings, lt::counters &counters)
+{
+    return std::make_unique<customDiskIOThread>(lt::default_disk_io_constructor(ioContext, settings, counters));
+}
+
+customDiskIOThread::customDiskIOThread(std::unique_ptr<libtorrent::disk_interface> nativeDiskIOThread)
+    : m_nativeDiskIO {std::move(nativeDiskIOThread)}
+{
+}
+
+lt::storage_holder customDiskIOThread::new_torrent(const lt::storage_params &storageParams, const std::shared_ptr<void> &torrent)
+{
+    lt::storage_holder storageHolder = m_nativeDiskIO->new_torrent(storageParams, torrent);
+
+    const QString savePath = Utils::Fs::expandPathAbs(QString::fromStdString(storageParams.path));
+    m_storageData[static_cast<lt::storage_index_t>(storageHolder)] = {
```

Comment:
are you sure you need the `static_cast<>` here? `storage_holder` is supposed to have an *implicit* conversion operator to `storage_index_t`